app/controllers/curation_concerns/generic_works_controller.rb

```diff
@@ -30,6 +31,14 @@ def identifiers
     end
   end
  
+  def assign_visibility
+    byebug
+    if (params["generic_work"]["visibility"]=="true")
```

Comment:
I suggest using a different parameter name for the form to avoid confusion as to the valid values of visibility.